packages/amplify-graphql-relational-transformer/src/resolvers.ts

```diff
@@ -8,6 +8,57 @@ import { getSortKeyFields } from './schema';
 import { HasManyDirectiveConfiguration } from './types';
 import { getConnectionAttributeName, getObjectPrimaryKey } from './utils';
 
+export const updateTableForReferencesConnection = (
+  config: HasManyDirectiveConfiguration, // TODO: Add support for HasOneDirectiveConfiguration
+  ctx: TransformerContextProvider
+): void => {
+  const { field, fieldNodes, indexName: incomingIndexName, object, references, relatedType } = config;
+
+  if (incomingIndexName) {
+    // TODO: log warning or throw that indexName isn't supported for DDB references
+    // Ideally validate this further up the chain.
+  }
+
+  if (references.length < 1) {
+    throw new Error( // TODO: better error message
+      'references should not be empty here'
+    )
+  }
+
+  const mappedObjectName = ctx.resourceHelper.getModelNameMapping(object.name.value);
+  const indexName = `gsi-${mappedObjectName}.${field.name.value}`;
+  config.indexName = indexName;
+
+  const table = getTable(ctx, relatedType);
```

Comment:
naming nit
```suggestion
  const relatedTable = getTable(ctx, relatedType);
```

Reply:
Good call, addressed in 0d3bce66584dd652cabba1ecc7987cdae3de3154